CONTRIBUTING.md

```diff
@@ -37,6 +37,46 @@ Only required if dealing with manuals, `gh-pages` or releases:
 3. Make sure that everything works on the current platform by running `make test`
 4. [Run local CI tests](#running-local-ci-tests) to verify functionality on supported platforms `bundle exec kitchen verify --test-base-path="$PWD/.ci/integration"`.
 
+### Code style
+
+New features and changes should aim to be as clear, concise, simple, and consistent
+
+1. clear: make it as obvious as possible what the code is doing
+ 
+2. concise: your PR should be as few characters (not just lines) of changes as _reasonable_. 
+   However, generally choose clarity over being concise.  
+   Clarity and conciseness can be in conflict with each other. But
+   it's more important for the code to be understandable than for it to be small.
+   Therefore favor writing clear code over making shorter diffs in your PRs.
+ 
+3. simple: this dovetails with the previous two items. 
+   git-secret is a security product, so it's best to have the code be easy to understand.
+   This also aids future development and helps minimize bugs.
+
+4. consistent: Write code that is consistent with the surrounding code and the rest of the git-secret code base.
+   Every code base has its own conventions and style that develop and accrete over time.
+
+   Consistency also means that the inputs and outputs of git-secret should be as consistent as reasonable
+   with related unix and git tools, and follow the 'rule of least surprise', 
+   also known as the 'principle of least astonishment': <https://en.wikipedia.org/wiki/Principle_of_least_astonishment>
+
+We wrote this to clarify our thinking about how git-secret should be written.  Of course, these are philosophical goals, 
+not necessities for releasing code, so balancing these four ideals _perfectly_ is both unwarranted and impossible.
+
+### Writing PRs
+
+If you're planning a large change to git-secret (for example, a lot of lines/characters of diffs, affecting multiple commands, 
```

Comment:
It should be `git-secret` for consistency

Reply:
I don't understand, isn't that what it says?

Reply:
oh I see, with backticks